crates/fluvio-protocol-derive/src/util.rs

```diff
@@ -1,75 +1,256 @@
-use syn::{Attribute, Lit, LitStr, Meta, MetaNameValue, NestedMeta};
+use proc_macro2::Span;
+use syn::{
+    meta::ParseNestedMeta, punctuated::Punctuated, Attribute, Expr, ExprLit, ExprPath, ExprUnary,
+    Lit, LitStr, Meta, MetaList, MetaNameValue, Token, UnOp,
+};
 
-pub(crate) fn find_attr(attrs: &[Attribute], name: &str) -> Option<Meta> {
-    attrs.iter().find_map(|a| {
-        if let Ok(meta) = a.parse_meta() {
-            if meta.path().is_ident(name) {
-                Some(meta)
-            } else {
-                //println!("attr name: {}",meta.name());
-                None
+use crate::ast::prop::PropAttrsType;
+
+/// Parses the specified attributes from a `syn::Attribute` iterator.
+/// # Arguments
+/// * `attrs` - `&[Attribute]` iterator
+/// * `attr_ident` - The path ident to search for
+/// * `meta` - a return variable that will give us the parse_nested_meta of each element
+/// * `field` - nested path to search for
+/// * `opt` - mutable variable to save data into
```

Comment:
The `mutable variable to save data into` comment contradicts with the only occurrence `if $opt.is_none()`.
Handling duplicates of parsed attributes should be part of `parse_attributes` operation.

Reply:
Ah yes.. the option is actually just used as a way to check if the field is already used. Updated the docs.